ci-operator/step-registry/hypershift/mce/power/create/hcp/hypershift-mce-power-create-hcp-commands.sh

```diff
@@ -0,0 +1,196 @@
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: ${FS_VOLUME_SIZE}
+  mirrorRegistryRef:
+    name: mirror-config
+  osImages:
+    - openshiftVersion: "${CLUSTER_VERSION}"
+      version: $(echo "$OS_IMAGES" | jq -r '.[] | select(.cpu_architecture == "ppc64le").version')
+      url: $(echo "$OS_IMAGES" | jq -r '.[] | select(.cpu_architecture == "ppc64le").url')
+      cpuArchitecture: ppc64le
+EOF
+
+oc wait --timeout=5m --for=condition=DeploymentsHealthy agentserviceconfig agent
+echo "$(date) AgentServiceConfig ready"
+
+set +x
+# Setting up pull secret with brew token
+oc extract secret/pull-secret -n openshift-config --to=/tmp --confirm
+brewToken="${AGENT_POWER_CREDENTIALS}/brew-token"
+cat /tmp/.dockerconfigjson | jq --arg brew_token "$(cat ${brewToken})" '.auths += {"brew.registry.redhat.io": {"auth": $brew_token}}' > /tmp/pull-secret
+PULL_SECRET=/tmp/pull-secret
+set -x
+
+# Creating agent hosted cluster manifests
+echo "$(date) Creating agent hosted cluster manifests"
+oc create ns ${HOSTED_CONTROL_PLANE_NAMESPACE}
+mkdir /tmp/hc-manifests
+
+ICSP_COMMAND=""
+if [[ $ENABLE_ICSP == "true" ]]; then
+  ICSP_COMMAND=$(echo "--image-content-sources ${SHARED_DIR}/mgmt_icsp.yaml")
+fi
+
+SSH_PUB_KEY_FILE="${AGENT_POWER_CREDENTIALS}/ssh-publickey"
+
+${HYPERSHIFT_CLI_NAME} create cluster agent ${ICSP_COMMAND} \
+    --name=${HOSTED_CLUSTER_NAME} \
+    --namespace=${CLUSTERS_NAMESPACE} \
+    --pull-secret=${PULL_SECRET} \
+    --agent-namespace=${HOSTED_CONTROL_PLANE_NAMESPACE} \
+    --base-domain=${HYPERSHIFT_BASE_DOMAIN} \
+    --api-server-address=api.${HOSTED_CLUSTER_NAME}.${HYPERSHIFT_BASE_DOMAIN} \
+    --ssh-key=${SSH_PUB_KEY_FILE}\
+    --release-image=${OCP_IMAGE_MULTI} \
+    --control-plane-availability-policy=${CP_AVAILABILITY_POLICY} \
+    --infra-availability-policy ${HYPERSHIFT_INFRA_AVAILABILITY_POLICY} \
+    --render > /tmp/hc-manifests/cluster-agent.yaml
+
+# Split the manifest to replace routing strategy of various services
+csplit -f /tmp/hc-manifests/manifest_ -k /tmp/hc-manifests/cluster-agent.yaml /---/ "{6}"
+
+# Service strategy to replace
+printf "  - service: APIServer
+    servicePublishingStrategy:
+      type: LoadBalancer
+  - service: OAuthServer
+    servicePublishingStrategy:
+      type: Route
+  - service: OIDC
+    servicePublishingStrategy:
+      type: None
+  - service: Konnectivity
+    servicePublishingStrategy:
+      type: Route
+  - service: Ignition
+    servicePublishingStrategy:
+      type: Route
+  - service: OVNSbDb
+    servicePublishingStrategy:
+      type: Route
+" > /tmp/hc-manifests/replacement.yaml
+
+for file in /tmp/hc-manifests/manifest_*
+do
+    if grep -q 'kind: HostedCluster' "$file"
+    then
+        yq eval-all -i 'select(fileIndex==0).spec.services = select(fileIndex==1) | select(fileIndex==0)' "$file" "/tmp/hc-manifests/replacement.yaml"
+    fi
+
+    if grep -q 'kind: NodePool' "$file"
+    then
+        rm -f "$file"
+    fi
```

Comment:
We can achieve this by setting --node-pool-replicas.
```
 --node-pool-replicas int32        If set to 0 or greater, NodePools will be created with that many replicas. If set to less than 0, no NodePools will be created.
```

Reply:
Oh ok nice, did nt know about this before, thanks!